src/screens/RecoverySeed/RecoverySeedScreen.tsx

```diff
@@ -95,12 +95,13 @@ class RecoverySeedScreen extends Component<Props, State> {
       {
         isLoading: true,
       },
-      async () => {
+      () => {
         setTimeout(async () => {
           try {
             const keyPair = await mnemonicToKeyPair(mnemonic.join(' '));
-            onSubmit(keyPair, mnemonic);
-            this.setState({ isLoading: false });
+            this.setState({ isLoading: false }, () => {
+              onSubmit(keyPair, mnemonic);
+            });
```

Comment:
Why do you want to be sure that spinner is gone to call onSubmit function?

Reply:
without this there was a warning `Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method.` because isLoading was updated after redirect to next screen